src/extensions/http_api/client.js

```diff
@@ -13,230 +13,256 @@ const { Cookie } = require('tough-cookie')
-
-// RESPONSE INFORMATION
-let response = null
-let responseCookies = {}
-
-/**
- * Resets the client.
- */
-exports.reset = () => {
-    body = null
-    bodyType = null
-    headers = null
-    query = null
-
-    cookies = []
-    cookieJar = null
-
-    response = null
-    responseCookies = {}
-}
-
 /**
- * Sets request json body.
+ * Http Api Client extension.
  *
- * @param {Object} payload
+ * @class
  */
-exports.setJsonBody = payload => {
-    bodyType = BODY_TYPE_JSON
-    body = payload
-}
+class HttpApiClient {
+    constructor() {
+        // REQUEST INFORMATION
+        this.body = null
+        this.bodyType = null
+        this.headers = null
+        this.query = null
+        this.cookies = []
+        this.cookieJar = null
+
+        // RESPONSE INFORMATION
+        this.response = null
+        this.responseCookies = {}
+    }
 
-/**
- * Sets request form body.
- *
- * @param {Object} payload
- */
-exports.setFormBody = payload => {
-    bodyType = BODY_TYPE_FORM
-    body = payload
-}
+    /**
+     * Resets the client.
+     */
+    reset() {
+        this.body = null
+        this.bodyType = null
+        this.headers = null
+        this.query = null
 
-/**
- * Clears current request body
- */
-exports.clearBody = () => {
-    body = null
-    bodyType = null
-}
+        this.cookies = []
+        this.cookieJar = null
 
-/**
- * Sets request query parameters.
- *
- * @param {Object} _query
- */
-exports.setQuery = _query => {
-    query = _query
-}
+        this.response = null
+        this.responseCookies = {}
+    }
 
-/**
- * Sets request headers.
- *
- * @param {Object} _headers
- */
-exports.setHeaders = _headers => {
-    headers = _headers
-}
+    /**
+     * Sets request json body.
+     *
+     * @param {Object} payload
+     */
+    setJsonBody(payload) {
+        this.bodyType = BODY_TYPE_JSON
+        this.body = payload
+    }
 
-/**
- * Sets a single request header.
- *
- * @param {string} key
- * @param {string} value
- */
-exports.setHeader = (key, value) => {
-    headers = headers || {}
-    headers[key] = value
-}
+    /**
+     * Sets request form body.
+     *
+     * @param {Object} payload
+     */
+    setFormBody(payload) {
+        this.bodyType = BODY_TYPE_FORM
+        this.body = payload
+    }
 
-/**
- * Clears current request headers.
- */
-exports.clearHeaders = () => {
-    headers = null
-}
+    /**
+     * Clears current request body
+     */
+    clearBody() {
+        this.body = null
+        this.bodyType = null
+    }
 
-/**
- * Enables cookie jar.
- */
-exports.enableCookies = () => {
-    if (cookieJar !== null) return
+    /**
+     * Sets request query parameters.
+     *
+     * @param {Object} _query
+     */
+    setQuery(_query) {
+        this.query = _query
+    }
 
-    cookieJar = request.jar()
-    cookieJar._jar.rejectPublicSuffixes = false
-}
+    /**
+     * Sets request headers.
+     *
+     * @param {Object} _headers
+     */
+    setHeaders(_headers) {
```

Comment:
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

Reply:
ok